modules/mono/editor/bindings_generator.cpp

```diff
@@ -1483,9 +1483,9 @@ Error BindingsGenerator::_generate_cs_type(const TypeInterface &itype, const Str
 
 		output << MEMBER_BEGIN "public static GodotObject " CS_PROPERTY_SINGLETON "\n" INDENT1 "{\n"
 			   << INDENT2 "get\n" INDENT2 "{\n" INDENT3 "if (singleton == null)\n"
-			   << INDENT4 "singleton = " C_METHOD_ENGINE_GET_SINGLETON "(typeof("
-			   << itype.proxy_name
-			   << ").Name);\n" INDENT3 "return singleton;\n" INDENT2 "}\n" INDENT1 "}\n";
+			   << INDENT4 "singleton = " C_METHOD_ENGINE_GET_SINGLETON "(\""
+			   << itype.cname
```

Comment:
Shouldn't it be `itype.name` like we use in the static `NativeName` field?

```suggestion
			   << itype.name
```

Reply:
I am actually not sure whats the difference between those two, beyond that one is a `String` and the other a `StringName` I picked `cname` because you used `type_cname` in your PR and that was the closest thing available here and it gave the correct result.

Reply:
I think they are the same, probably to avoid converting between `String` and `StringName` multiple times.

In my PR, the `Engine::has_singleton` method takes a `StringName` so I used `cname`. In this case, since the `<<` operator takes a `String` it may be better to use `name` and avoid the conversion.

Reply:
Well, I tried both and they generate the exact same code, so I switched to avoid the unnecessary conversion and keep it in sync with the `NativeName`.